core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -221,18 +277,29 @@ void removeBuddyInfoForRemovedUser(User user, Buddy buddy)
 			// Buddy request was accepted
 			removeMessagesSentByBuddy(user, buddy);
 
-			sendDropBuddyMessage(null, buddy.getUserAnonymizedID(), buddy.getNickname(), user.getUserAnonymizedID(),
-					Optional.empty(), DropBuddyReason.USER_ACCOUNT_DELETED);
+			// Send message to "self", to notify the user about the removed buddy user
+			UUID buddyUserAnonymizedID = getUserAnonymizedIDForBuddy(buddy, "buddy relationship is established");
+			sendDropBuddyMessage(null, buddyUserAnonymizedID, buddy.getNickname(), Optional.empty(),
+					DropBuddyReason.USER_ACCOUNT_DELETED, user.getNamedMessageDestination());
 		}
 		else if (buddy.getSendingStatus() != Status.REJECTED && buddy.getReceivingStatus() != Status.REJECTED)
 		{
 			// Buddy request was not accepted or rejected yet
-			sendBuddyConnectResponseMessage(null, buddy.getUserAnonymizedID(), buddy.getNickname(), user.getUserAnonymizedID(),
+			// Send message to "self", as if the requested user declined the buddy request
+			UUID buddyUserAnonymizedID = buddy.getUserAnonymizedID().orElse(null); //
+			sendBuddyConnectResponseMessage(null, buddyUserAnonymizedID, buddy.getNickname(), user.getUserAnonymizedID(),
 					buddy.getID(), Status.REJECTED, getDropBuddyMessage(DropBuddyReason.USER_ACCOUNT_DELETED, Optional.empty()));
 		}
 		removeBuddy(user, buddy);
 	}
 
+	private UUID getUserAnonymizedIDForBuddy(Buddy buddy, String state)
```

Comment:
I always see the same `state` string in the code. Should it not discriminate between the different occasions?